h2o-core/src/main/java/water/Job.java

```diff
@@ -259,6 +259,7 @@ public final void update( final long newworked, final String msg) {
     }.invoke(LIST);
     // 4 - Fire off the FJTASK
     H2O.submitTask(fjtask);
+    H2O.updateNotIdle();
```

Comment:
This will only track jobs started on single node. I know we typically communicate with the same node but I can see optimization of starting jobs on different nodes to distributed the single node workload most our algos have. 

Just a suggestion, but I would maybe consider adding an atomic counter of running jobs instead. We can increase it when jobs starts and decrease it when it ends. The last job (the one who sets the atomic counter to 0) can set the last active time. We can collect these from all nodes upon the isIdle request or add this information to the heartbeat.

Actually now that I think of it, you already walk a list of jobs when you get isIdle call, that already has the information wee need. The only problem is it might be removed by removeAllKeys call, but you already hooked it. Can't we just remove job hooks altogether and just walk the job list every time you get is idle call?